File: Palindrome/C/palindrome.c
#include <stdio.h>
#include <stdbool.h>

bool isPalindrome(int x) {
    if (x < 0 || (x % 10 == 0 && x != 0)) return false;

    int temp = x;
    long long reverse = 0; // will contain the reversed number

    while(temp > 0) {
        int digit = temp % 10; // extracts the last digit from temp
        reverse = reverse * 10 + digit; // adds digit to reverse shifting the old ones to the left
        temp /= 10; // removes last digit from temp
    }
    return reverse == x;
}

int main(void) {

    bool result = isPalindrome(121);
    if(result) {
        printf("True.\n");
    } else {
        printf("False.\n");
    }

    return 0;
}

File: Palindrome/Python/palindrome.py
class Solution(object):
    def isPalindrome(self, x: int) -> bool:
        """
        :type x: int
        :rtype: bool
        """
        if x < 0:
            return False
        else:
            x = str(x)
            if x == x[::-1]:
                return True
            else:
                return False

if __name__ == "__main__":
    solution = Solution()
    x = int(input("Enter a number to check: "))
    result = solution.isPalindrome(x)
    if result:
        print("The number is palindrome")
    else:
        print("The number is not palindrome")
